cvmfs/swissknife_sync.cc

```diff
@@ -559,6 +560,17 @@ bool swissknife::CommandSync::ReadFileChunkingArgs(
 int swissknife::CommandSync::Main(const swissknife::ArgumentList &args) {
   string start_time = GetGMTimestamp();
 
+  // Spawn monitoring process (watchdog)
+  char watchdog_path[128];
+  std::string timestamp = GetGMTimestamp("%Y.%m.%d-%H.%M.%S");
+  int path_size = snprintf(watchdog_path, sizeof(watchdog_path),
```

Comment:
We should also create `watchdog_path` and make sure it's writable (otherwise crash right away).

Reply:
Hummm, we should not create the file, otherwise we will create a lot of files every time we use this code path.

We should create the file, iff some crash happens, not always. Am I wrong?

If we agree on that, then, it should be responsibility of the Watchdog to create the file, not of us invoking the watchdog.

Maybe, a sensible solution would be to check if the watchdog can create files and write in the stacktrace directory.

We can add it, but I am not sure what happens if the watchdog cannot write in the directory, do we crash? I am afraid we might break some installation.

I am going ahead and just create the directory that will contains the stacktrace file, it seems the most sensible solution.

Reply:
Looks good to me know!  I think the important part was creating the parent path, which is now done.